apps/oxlint/src/command/lint.rs

```diff
@@ -211,64 +211,178 @@ impl FromStr for OutputFormat {
-    pub oxc_plugin: bool,
+    #[bpaf(long("disable-oxc-plugin"), flag(Enable::Disable, Enable::NotSet), hide_usage)]
+    pub oxc_plugin: Enable,
 
     /// Disable TypeScript plugin, which is turned on by default
-    #[bpaf(long("disable-typescript-plugin"), flag(false, true), hide_usage)]
-    pub typescript_plugin: bool,
+    #[bpaf(long("disable-typescript-plugin"), flag(Enable::Disable, Enable::NotSet), hide_usage)]
+    pub typescript_plugin: Enable,
 
     /// Enable the experimental import plugin and detect ESM problems.
     /// It is recommended to use along side with the `--tsconfig` option.
-    #[bpaf(switch, hide_usage)]
-    pub import_plugin: bool,
+    #[bpaf(flag(Enable::Enable, Enable::NotSet), hide_usage)]
+    pub import_plugin: Enable,
 
     /// Enable the experimental jsdoc plugin and detect JSDoc problems
-    #[bpaf(switch, hide_usage)]
-    pub jsdoc_plugin: bool,
+    #[bpaf(flag(Enable::Enable, Enable::NotSet), hide_usage)]
+    pub jsdoc_plugin: Enable,
 
     /// Enable the Jest plugin and detect test problems
-    #[bpaf(switch, hide_usage)]
-    pub jest_plugin: bool,
+    #[bpaf(flag(Enable::Enable, Enable::NotSet), hide_usage)]
+    pub jest_plugin: Enable,
 
     /// Enable the Vitest plugin and detect test problems
-    #[bpaf(switch, hide_usage)]
-    pub vitest_plugin: bool,
+    #[bpaf(flag(Enable::Enable, Enable::NotSet), hide_usage)]
+    pub vitest_plugin: Enable,
 
     /// Enable the JSX-a11y plugin and detect accessibility problems
-    #[bpaf(switch, hide_usage)]
-    pub jsx_a11y_plugin: bool,
+    #[bpaf(flag(Enable::Enable, Enable::NotSet), hide_usage)]
+    pub jsx_a11y_plugin: Enable,
 
     /// Enable the Next.js plugin and detect Next.js problems
-    #[bpaf(switch, hide_usage)]
-    pub nextjs_plugin: bool,
+    #[bpaf(flag(Enable::Enable, Enable::NotSet), hide_usage)]
+    pub nextjs_plugin: Enable,
 
     /// Enable the React performance plugin and detect rendering performance problems
-    #[bpaf(switch, hide_usage)]
-    pub react_perf_plugin: bool,
+    #[bpaf(flag(Enable::Enable, Enable::NotSet), hide_usage)]
+    pub react_perf_plugin: Enable,
 
     /// Enable the promise plugin and detect promise usage problems
-    #[bpaf(switch, hide_usage)]
-    pub promise_plugin: bool,
+    #[bpaf(flag(Enable::Enable, Enable::NotSet), hide_usage)]
+    pub promise_plugin: Enable,
 
     /// Enable the node plugin and detect node usage problems
-    #[bpaf(switch, hide_usage)]
-    pub node_plugin: bool,
+    #[bpaf(flag(Enable::Enable, Enable::NotSet), hide_usage)]
+    pub node_plugin: Enable,
 
     /// Enable the security plugin and detect security problems
-    #[bpaf(switch, hide_usage)]
-    pub security_plugin: bool,
+    #[bpaf(flag(Enable::Enable, Enable::NotSet), hide_usage)]
+    pub security_plugin: Enable,
+}
+
+#[derive(Default, Debug, Clone, Copy, PartialEq, Eq)]
+#[allow(clippy::enum_variant_names)]
+pub enum Enable {
```

Comment:
I'm finding this name a bit confusing, maybe we could differentiate from the `Enable::Enable` member? perhaps: `Enablement`?

Reply:
Thanks for the feedback. I'll rename it in the morning.

Reply:
I renamed it to `OverrideToggle` and added some doc comments.